evalml/tests/automl_tests/test_automl.py

```diff
@@ -5480,3 +5490,191 @@ def test_holdout_set_results_and_rankings(caplog, AutoMLTestEnv):
+        use_recommendation=True,
+        include_recommendation=["MedianAE", "MaxError"],
+        exclude_recommendation=["MAE"],
+    )
+    automl.search()
+
+    scores = automl.get_recommendation_scores(
+        priority="MaxError",
+        priority_weight=0.7,
+    )
+    for score in scores.values():
+        assert 0 <= score <= 100
+    for pipeline_id in scores.keys():
+        assert isinstance(pipeline_id, int)
+
+    scores = automl.get_recommendation_scores(use_pipeline_names=True)
+    for pipeline_name in scores.keys():
+        assert isinstance(pipeline_name, str)
+
+
+def test_recommendation_score_argument_errors(X_y_binary, caplog):
+    caplog.clear()
+    X, y = X_y_binary
+
+    with pytest.raises(
+        ValueError,
+        match="Objectives to include from the recommendation score should be a list",
+    ):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation="Not a list",
+        )
+    with pytest.raises(
+        ValueError,
+        match="Objectives to exclude from the recommendation score should be a list",
+    ):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            exclude_recommendation="Not a list",
+        )
+    with pytest.raises(ObjectiveNotFoundError, match="not a valid Objective!"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation=["Does not exist"],
+        )
+    with pytest.raises(ValueError, match="not defined for binary"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation=["R2"],
+        )
+    with pytest.raises(ValueError, match="Cannot exclude objective"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            exclude_recommendation=["Precision"],
+        )
+    AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        use_recommendation=True,
+        include_recommendation=["F1"],
+    )
+    assert "already one of the default objectives" in caplog.text
+
+
+def test_recommendation_include_non_optimization(X_y_binary):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        use_recommendation=True,
+        include_recommendation=["Recall"],
+    )
+    assert "Recall" in automl.recommendation_objectives
+    assert get_objective("Recall") in automl.additional_objectives
+
+
+@pytest.mark.parametrize("imbalanced_data", [True, False])
+def test_use_recommendation_score_imbalanced(
```

Comment:
I reviewed the code...why are we treating imbalanced data differently?

Reply:
This was discussed in the design doc - it's exclusively used in the multiclass classification case, when selecting whether to use AUC Micro or AUC weighted, because micro averages are better for imbalanced datasets.